frontend/components/Icon/Icon.js

```diff
@@ -0,0 +1,29 @@
+import React from "react";
+import PropTypes from "prop-types";
+import styled from "styled-components";
+import { styles } from "./styles";
+import * as icons from "../../svgs";
+
+const StyledIcon = styled.i`
+  ${styles};
+`;
+
+function Icon({ icon, ...rest }) {
+  const Svg = icons[icon];
+  if (!Svg) {
+    return null;
+  }
+
+  return <StyledIcon {...rest} dangerouslySetInnerHTML={{ __html: Svg() }} />;
+}
+
+Icon.propTypes = {
+  size: PropTypes.oneOf(["xxs", "xs", "sm", "md", "lg", "xl", "full"]),
```

Comment:
Can you do propTypes like this for every component please. Also, are these right? I feel like this is a lot of icon sizes for a site that has 4 icons...